pkg/reconciler/dependencybuild/dependencybuild.go

```diff
@@ -788,6 +788,10 @@ func (r *ReconcileDependencyBuild) handleBuildPipelineRunReceived(ctx context.Co
 			if err != nil {
 				return reconcile.Result{}, err
 			}
+			err = r.createArtifacts(ctx, log, pr, db, deployed)
```

Comment:
I don't think we need to create `ArtifactBuild` objects for this case, we already have `RebuiltArtifact` as a marker that the objects were created.